pulsar-io/twitter/src/main/java/org/apache/pulsar/io/twitter/TwitterFireHose.java

```diff
@@ -72,7 +72,8 @@
 
     @Override
     public void open(Map<String, Object> config, SourceContext sourceContext) throws IOException {
-        TwitterFireHoseConfig hoseConfig = IOConfigUtils.loadWithSecrets(config, TwitterFireHoseConfig.class, sourceContext);
+        TwitterFireHoseConfig hoseConfigWithoutSecrets = TwitterFireHoseConfig.load(config);
```

Comment:
I don't quite understand why we need to separate this into two calls?  Seems to make things more complicated than necessary?

Reply:
Connectors can construct their config class in any manner they choose to. Mapping from the the Map<String, Object> using object factory is just one way of doing this. The existing method overloads both, when the function name really is about loading the config class with secrets. Hence the change

Reply:
I've changed the function name to better describe what the function is doing

Reply:
> Connectors can construct their config class in any manner they choose to. Mapping from the the Map<String, Object> using object factory is just one way of doing this.

Sure, but I would say the most common pattern would be the existing one.  Most, if not all connectors, will load a config class from a map where is is a one to one mapping of class field to key. I think it would be extremely uncommon for a connector not to follow that pattern and have some custom mapping scheme implemented. At most I would recommend just adding a helper method "IOConfigUtils.fillWithSecrets" that allows developers to do the "filling" of secrets separately if they want but I wouldn't recommend changing the paradigm for every connector.

Reply:
Closing this issue, since I now concur with Jerry's views.